### PR TITLE
feat: adding intermediate results to hmi-server

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
Adding intermediate results from pyciemss to hmi-server

Need to still test directly with pyciemss -> dev/1.6.0 will be going in shortly 
Need to confirm filtering is working as intended. Not taking out from queue unless the job id matches

### CODE REVIEW

File: packages/services/hmi-server/src/main/resources/application.properties [hunk 1]
```diff
@@ -45,8 +45,8 @@ skema/mp-rest/url=https://skema-py.staging.terarium.ai
 skema-rust/mp-rest/url=https://skema-rs.staging.terarium.ai
 skema-unified/mp-rest/url=https://skema-unified.staging.terarium.ai
 model-service/mp-rest/url=https://model-service.staging.terarium.ai
-%dev.data-service/mp-rest/url=http://${data-service-host:localhost}:${data-service-port:3020}
-%prod.data-service/mp-rest/url=http://${data-service}:3020
+%dev.data-service/mp-rest/url=http://${data-service-host:localhost}:${data-service-port:8001}
+%prod.data-service/mp-rest/url=http://${data-service}:8001
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/resources/application.properties:49]: I don't think this has been done, this is only on your local machine to do so.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java [hunk 2]
```diff
@@ -141,4 +151,16 @@ public Response createFromSimulationResult(
...
+
+	@GET
+	@Path("/{jobId}/partial-result")
+	@Produces(MediaType.SERVER_SENT_EVENTS)
+	@SseElementType(MediaType.APPLICATION_JSON)
+	@Tag(name = "Stream partial/intermediate simulation result associated with run ID")
+	public Publisher<CalibrationIntermediateResults> stream(
+		@PathParam("jobId") final String jobId
+	) {
+		// return Multi.createFrom().publisher(events);
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java:163]: Remove comment
```suggestion
```

File: packages/client/hmi-client/src/components/workflow/tera-simulation-workflow.vue [hunk 3]
```diff
@@ -424,6 +426,27 @@ function updateWorkflowNodeState(node: WorkflowNode, state: any) {
...
 	const value = (node.inputs[0].value?.[0] ?? 0) + Math.round(Math.random() * 10);
 	appendOutputPort(node, { type: 'number', label: value.toString(), value });
+
+	console.log('Checking MQ');
+	const auth = useAuthStore();
+	const ttt = new EventSourcePolyfill('/api/simulations/123/partial-result', {
+		headers: {
+			Authorization: `Bearer ${auth.token}`
+		}
+	});
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/tera-simulation-workflow.vue:436]: Please add a new utils in `/api/sse.ts` like we did with `/api/api.ts` to encapsulate the auth for now. So we can handle things like re-auth in the future.

File: packages/services/hmi-server/src/main/resources/application.properties (hunk 1 above)
Comment (Tom-Szendrey) [reply packages/services/hmi-server/src/main/resources/application.properties:49]: Yepp, this is a draft PR - just testing still was running into some errors had to run DB locally will revert when I open PR up

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java:163]: Will do

File: packages/client/hmi-client/src/components/workflow/tera-simulation-workflow.vue (hunk 3 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/tera-simulation-workflow.vue:436]: Good thinking!

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java [hunk 4]
```diff
@@ -141,4 +151,16 @@ public Response createFromSimulationResult(
...
+		return Multi.createFrom().publisher(events).select().where(event -> event.getJobId == jobId);
+	}
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java:165]: Is there ever a risk here where no `event.getJobId == jobId` evaluates to `true`? Will that return null or an exception - do we need to catch that and return a specific response in that case?

Comment (YohannParis) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java:183]: No, that's the purpose of SSE, just listening to the endpoint like a broadcast. We might never receive anything at all.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		this.jobId = jobId;
+	}
+
+	public void setProgress(float progress){
+		this.progress = progress;
+	}
+
+	public String getJobId(){
+		return this.jobId;
+	}
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java:26]: Isn't `@Data` supposed to handle those @dvince2?

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java [hunk 6]
```diff
@@ -141,4 +157,28 @@ public Response createFromSimulationResult(
...
+
+	@GET
+	@Path("/{jobId}/partial-result")
+	@Produces(MediaType.SERVER_SENT_EVENTS)
+	@SseElementType(MediaType.APPLICATION_JSON)
+	@Tag(name = "Stream partial/intermediate simulation result associated with run ID")
+	public Publisher<SimulationIntermediateResults> stream(
+		@PathParam("jobId") final String jobId
+	) {
+		return Multi.createFrom().publisher(partialSimulationResults).select().where(event -> event.getJobId().equals(jobId));
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java:169]: beautiful condition!

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java (hunk 5 above)
Comment (Tom-Szendrey) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java:26]: I thought so too but I couldnt use these functions until I created them myself 
Maybe I have the naming wrong for @data? Im not sure

Comment (dvince2) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java:26]: Strange... yeah the `@Data` annotation should take care of that.  You can use specific `@Setter` and `@Getter` on these fields as well too if not?

**Review by YohannParis — APPROVED** [review]
Good for me on the front-end, will let @dvince2 tell you for the back-end

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java (hunk 5 above)
Comment (dvince2) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/SimulationIntermediateResults.java:26]: image

image


I'm wondering if you're having an issue with Lombok/your IDE not showing the suggestions? When i blew away the setters/getters on your branch things looked ok to me

**Review by dvince2 — APPROVED** [review]

**Pull request merged**